src/modules/fw_pos_control/FixedwingPositionControl.cpp

```diff
@@ -94,6 +94,31 @@ FixedwingPositionControl::init()
 	return true;
 }
 
+float FixedwingPositionControl::getMaximumClimbRate()
+{
+
+	float weight_factor = 1.0f;
+
+	if (_param_weight_base.get() > 0.0f && _param_weight_gross.get() > 0.0f) {
+		weight_factor = math::constrain(_param_weight_base.get() / _param_weight_gross.get(), MIN_WEIGHT_RATIO,
+						MAX_WEIGHT_RATIO);
+	}
+
+	float climbrate_max = _param_fw_t_clmb_max.get();
+
+	const float density_min = _param_density_min.get();
+
+	if (density_min < AIR_DENSITY_STANDARD_ATMOS_1000_AMSL
```

Comment:
why 1000m for the constraint? just because the delta for climb rate is assumed negligible compared to sea level value if ceiling is less than 1000m? how negligible is it? (maybe you could make a quick example calc?) -- i see in general though it needs to be greater than sea level value for the following logic to not divide by zero